apps/nextjs/src/env/schema.mjs

```diff
@@ -5,22 +5,28 @@ import { z } from "zod";
  * Specify your server-side environment variables schema here.
  * This way you can ensure the app isn't built with invalid env vars.
  */
-export const serverSchema = z.object({
-  NODE_ENV: z.enum(["development", "test", "production"]),
-  GOOGLE_ID: z.string(),
-  GOOGLE_SECRET: z.string(),
-  AZURE_AD_CLIENT_ID: z.string(),
-  AZURE_AD_CLIENT_SECRET: z.string(),
-  AZURE_AD_TENANT_ID: z.string(),
-  NEXTAUTH_SECRET: z.string(),
-  NEXTAUTH_URL: z.preprocess(
-    (str) => process.env.VERCEL_URL ?? str,
-    process.env.VERCEL ? z.string() : z.string().url()
-  ),
-  // provided by vercel (therefore shouldn't throw during schema parsing)
-  VERCEL_URL: z.string().optional(),
-  PORT: z.string().transform((str) => parseInt(str, 10)),
-});
+export const serverSchema = z.lazy(() =>
+  z.object({
+    NODE_ENV: z.enum(["development", "test", "production"]),
+    GOOGLE_ID: z.string(),
+    GOOGLE_SECRET: z.string(),
+    AZURE_AD_CLIENT_ID: z.string(),
+    AZURE_AD_CLIENT_SECRET: z.string(),
+    AZURE_AD_TENANT_ID: z.string(),
+    NEXTAUTH_SECRET: z.string(),
+    PORT: z.string().transform((str) => parseInt(str, 10)),
+    HOST: z.string(),
+    NEXTAUTH_URL: z.preprocess(
+      (str) =>
+        process.env.VERCEL_URL
+          ? `https://${process.env.VERCEL_URL}/api/auth`
+          : str,
+      z.string().url()
```

Comment:
This is wrong, don't mind this. I [reverted](https://github.com/agreeto/agreeto/pull/6/commits/86043e58a34816226dec6707e74b2ef115a75d81#diff-f9af0df61ed17b9719323675fde2428270dcb82b9c09844562f8125fb7341b88R24).

Reply:
cool, yea the /api/auth part shouldn't be there.

also the VERCEL_URL env doesn't include the `https://` part so it's not a valid zod url, hence the optional string / string+url

Reply:
Okay, and `NEXTAUTH_URL` works also without the procotol? Or do we still have to add that?

Reply:
yes it works without protocol